src/platform/renesas/DiagnosticDataProviderImpl.cpp

```diff
@@ -93,6 +93,42 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetBootReason(BootReasonType & bootReason
     return err;
 }
 
+CHIP_ERROR DiagnosticDataProviderImpl::GetNetworkInterfaces(NetworkInterface ** netifpp)
+{
+    struct netif * net_interface;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    NetworkInterface * ifp = new NetworkInterface();
```

Comment:
Are you sure that this gets only called once?

How about a static global somewhere (or an array?)

Reply:
I think they are deleted in ReleaseNetworkInterfaces. Making them static would reserve ram permanently for a structure that may be read once or never during the lifetime of the bridge. 

The dynamic memory allocation of course also has a cost (slightly more cpu cost, small chance of fragemntation) but I think it's negligible.